gtwrap/matlab_wrapper/wrapper.py

```diff
@@ -137,6 +139,35 @@ def _insert_spaces(self, x, y):
         """
         return x + '\n' + ('' if y == '' else '  ') + y
 
+    @staticmethod
+    def _expand_default_arguments(method, save_backup=True):
```

Comment:
I was thinking of something simpler where you just iterate over the arguments and you create methods with more and more arguments. That would just involve adding another loop for the method generation code.

Reply:
One reason I wanted to do it this way with the copying is because the method generation code occurs in 4-5 places since constructors, methods, static methods, global functions, etc. are all handled differently.  This way, we only have to modify the code in one place instead of 4-5 which I figured would make maintainability easier.

I did consider making this iterator-style using `yield` etc instead of returning a list, but there's a lot of places in the code that use `isinstance(methods, list)` and also joining together iterators (e.g. in `wrap_namespace`:  `[func for func in namespace.content if ...]`) that would make this confusing.

Reply:
Hmm then this could be called in each of those methods. I guess that's what you are doing right now?

Reply:
ya, well right now each of the method generation functions are (except ctor which is handled separately) already using the overload resolution so I call this function from the overload resolution method `_group_methods`.

We could instead call something similar from each of the methods that generate function wraps but it just seems a little bit harder to maintain to me, but I can do that if you like.

Reply:
I think this is fine. The comments were more for my own clarification.